app/src/main/java/com/duckduckgo/app/global/image/GlobalGlideModule.kt

```diff
@@ -16,8 +16,54 @@
 
 package com.duckduckgo.app.global.image
 
+import android.content.Context
+import android.os.Build
+import com.bumptech.glide.Glide
+import com.bumptech.glide.Registry
 import com.bumptech.glide.annotation.GlideModule
+import com.bumptech.glide.integration.okhttp3.OkHttpUrlLoader
+import com.bumptech.glide.load.model.GlideUrl
 import com.bumptech.glide.module.AppGlideModule
+import com.duckduckgo.app.browser.certificates.rootstore.IsrgRootX1
+import com.duckduckgo.app.browser.certificates.rootstore.IsrgRootX2
+import okhttp3.OkHttpClient
+import okhttp3.tls.HandshakeCertificates
+import timber.log.Timber
+import java.io.InputStream
+import java.security.cert.X509Certificate
 
 @GlideModule
-class GlobalGlideModule : AppGlideModule()
+class GlobalGlideModule : AppGlideModule() {
+    override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N_MR1) {
+            try {
+                Timber.d("Registering OkHttp-based ModelLoader for GlideUrl")
+
+                val isrgRootX1 = IsrgRootX1(context)
+                val isrgRootX2 = IsrgRootX2(context)
+
+                val handshakeCertificates = HandshakeCertificates.Builder()
+                    .addTrustedCertificate(isrgRootX1.certificate() as X509Certificate)
+                    .addTrustedCertificate(isrgRootX2.certificate() as X509Certificate)
+                    .addPlatformTrustedCertificates()
+                    .build()
+
+                val okHttpClient = OkHttpClient.Builder()
+                    .sslSocketFactory(handshakeCertificates.sslSocketFactory())
```

Comment:
`.sslSocketFactory` is marked as deprecated with a suggestion to use another function. 

Should we be using the suggested alternative? Or if we can't for some reason, can you suppress this warning to indicate we are knowingly using it?

Reply:
I was intentional about using that deprecated function. I prefer not to manually pass myself the system `TrusManager` to avoid having an issue there. The only difference is that the deprecated function uses reflection, which it does not impose any performance issues for this one instance.

Reply:
If it's intentional, let's make it _obviously_ intentional; whereas now in code it looks accidental. Suppressing that warning (and perhaps a comment explaining that decision) would help indicate it was intentional.

Reply:
Makes sense, I will suppress the warning

Reply:
It's deprecated for a reason, you should be probably be passing in `handshakeCertificates.trustManager()`.

It will work on Android via StandardAndroidSocketAdapter, but would fail if you happened to have the GMS Play provider or conscrypt installed.

Reply:
oh, nice. Missed the API in `handshakeCertificates` where we can get the `trusManager`. I thought I was gonna have to  go through the `TrustManagerFactory`, `init` with `null` keystore and all.

I will use your suggestion then.

Now that I have you here, what was the reason to deprecate it, just curious.